lib/ansible/modules/monitoring/grafana_datasource.py

```diff
@@ -0,0 +1,469 @@
+    description:
+      - Grafana API user
+  grafana_password:
+    required: false
+    default: admin
+    description:
+      - Grafana API password
+  grafana_api_key:
+    required: false
+    description:
+      - Grafana API key
+      - If set, grafana_user and grafana_password will be ignored
+  database:
+    required: false
+    description:
+      - Name of the database for the datasource.
+      - This options is required when the ds_type is influxdb or elasticsearch (index name)
+  user:
+    required: false
+    description:
+      - Datasource login user for influxdb datasources
+  password:
+    required: false
+    description:
+      - Datasource password
+  basic_auth_user:
+    required: false
+    description:
+      - Datasource basic auth user.
+      - Setting this option with basic_auth_password will enable basic auth.
+  basic_auth_password:
+    required: false
+    description:
+      - Datasource basic auth password, when basic auth is true
+  with_credentials:
+    required: false
+    default: false
+    description:
+      - Whether credentials such as cookies or auth headers should be sent with cross-site requests.
+  tls_client_cert:
+    required: false
+    description:
+      - client tls certificate.
+      - If tls_client_cert and tls_client_key are set, this will enable tls auth.
+      - begins with ----- BEGIN CERTIFICATE -----
+  tls_client_key:
+    required: false
+    description:
+      - client tls private key
+      - befins with ----- BEGIN RSA PRIVATE KEY -----
+  tls_ca_cert:
+    required: false
+    description:
+      - tls ca certificate for self signed certificates.
+      - only used when tls_client_cert and tls_client_key are set
+  is_default:
+    required: false
+    default: false
+    description:
+      - Make this datasource the default one
+  org_id:
+    required: false
+    description:
+      - Grafana Organisation ID in which the datasource should be created
+      - Not used when grafana_api_key is set, because the grafana_api_key only belong to one organisation.
+    default: 1
+  state:
+    required: false
+    default: present
+    choices: [present, absent]
+    description:
+      - Status of the datasource
+  es_version:
+    required: false
+    default: 5
+    description:
+      - Elasticsearch version (for ds_type = elasticsearch only)
+  time_field:
+    required: false
+    default: timestamp
+    description:
+      - Name of the time field in elasticsearch ds
+      - For example @timestamp
+  time_interval:
+    required: false
+    description:
+      - Minimum group by interval for influxdb or elasticsearch datasources
+      - for example '>10s'
+  tsdb_version:
+    required: false
+    choices: [1, 2, 3]
+    description:
+      - opentsdb version (1 for <= 2.1, 2 for ==2.2, 3 for ==2.3)
+    default: 1
+  tsdb_resolution:
+    required: false
+    choices: [1, 2]
+    description:
+      - opentsdb time resolution (1 for second, 2 for millisecond)
```

Comment:
Can we set the choices to `[seconds, milliseconds]` then convert it before passing the data to grafana?

Reply:
ok, I've changed the available choices then convert it for grafana api.